### PR TITLE
[CI] fix Ceph functional tests

### PR DESCRIPTION
**Pull request by r-f-g**
- add Cache class with
  - enable/disable function
  - enabled property -> to show actual state
  - run_cache -> to temporary change cache state

### CODE REVIEW

**Review by auria — DISMISSED** [review]
+1, LGTM.

**Review by mkalcok — CHANGES_REQUESTED** [review]
I have only few comments regarding docstrings...otherwise LGTM

File: juju_verify/utils/cache.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+# version.
+#
+# juju-verify is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program. If not, see https://www.gnu.org/licenses/.
+"""Helper function to manage cache."""
```
[the remark this answers is not in the document]
Comment (r-f-g) [reply juju_verify/utils/cache.py:17]: What do you say to this?
"Utils for caching and subsequently managing the use of this memory."

[the remark this answers is not in the document]
Comment (mkalcok) [reply juju_verify/utils/cache.py:17]: "... use of Action Cache"?

File: juju_verify/utils/cache.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        self._cache.clear()
+
+    @property
+    def keys(self) -> List[Any]:
+        """Return cached keys."""
+        return list(self._cache.keys())
+
+
+class CacheManager:
+    """The Cache manager."""
```
[the remark this answers is not in the document]
Comment (r-f-g) [reply juju_verify/utils/cache.py:72]:
```python
class Cache:
    """Cache for storing outputs using specific keys

    Usage example:
    cache = Cache()

    def run_action(name: str) -> Any:
        if name in cache:
            return cache[name]

        return _run_action(name)

    run_action('test')  # action 'test' is executed
    run_action('test')  # action 'test' is not executed
    run_action('test')  # action 'test' is not executed
    """

class CacheManager:
    """A cache manager that determines when to use this cache.

    Usage example:
    cache_manager = CacheManager(enabled=True)
    cache = Cache()

    def run_action(name: str) -> Any:
        if cache_manager.enabled and name in cache:
            return cache[name]

        return _run_action(name)

    run_action('test')  # action 'test' is executed
    run_action('test')  # action 'test' is not executed
    with cache_manager(use_cache=False):
        run_action('test')  # action 'test' is executed
        
    run_action('test')  # action 'test' is not executed
    """
```

**Review by mkalcok — APPROVED** [review]

**Review by mkalcok — APPROVED** [review]

**Pull request merged**